front/app/components/PageForm/index.tsx

```diff
@@ -47,10 +48,33 @@ interface Props {
 const PageForm = ({ onSubmit, defaultValues, pageId }: Props) => {
   const { formatMessage } = useIntl();
   const page = useCustomPage({ customPageId: pageId });
-  const remoteFiles = useRemoteFiles({
-    resourceId: pageId,
-    resourceType: 'page',
-  });
+  const { data: remotePageFiles } = usePageFiles(
+    isNilOrError(page) ? undefined : page.id
+  );
+
+  const [files, setFiles] = React.useState<UploadFile[]>([]);
```

Comment:
Do we need the useEffect? Or is there a way we could initialize the data directly by moving the data.map into a function and calling it like:
`const [files, setFiles] = React.useState<UploadFile[]>(getUploadFiles(remotePageFiles.data) || []);`

```
          remotePageFiles.data.map(async (file) => {
            const uploadFile = convertUrlToUploadFile(
              file.attributes.file.url,
              file.id,
              file.attributes.name
            );
            return uploadFile;
          })
        )) as UploadFile[];
```

Reply:
When the files are updated in anyway, it should trigger the invalidation + rerender as well right? :thinking:

Reply:
We need it in this case because if we set it in the state directly, the `remotePageFiles` will not be immediately available and the state will not be updated afterwards. `useEffect` ensures the component stays in sync.